### PR TITLE
joyent/illumos-extra#52 upgrade OpenSSH to 8.6p1

### PR DESCRIPTION
**Pull request by arekinath**
Note that this drops the GSS key exchange patch (so the resulting OpenSSH will not support GSS key exchange, but it will still support GSS *auth* -- the two are separate things). Primarily this will mean that users who currently use GSS creds (e.g. KRB5) to log into machines will now be prompted to trust the host key of the server where they did not before, and some forms of credential delegation via GSS might no longer work (basic delegation seems to be fine via the GSS auth plugin).

The patch is being dropped due to the effort required to keep it up to date (it's by far the worst part of forward-porting the smartos patch set for OpenSSH) and the relatively small number of actual users. (I also personally opened libgssapi's code the other day and I would like to keep the use of GSS in critical services to a minimum.)

Testing done so far:
 * Built into a PI
 * Booted on vmware, verified that ssh service comes up, can SSH into the gz, gz can SSH out fine
 * Booted on real hardware (test CN) with existing zones, including a native-brand zone with kerberos GSS login support
 * Verified that test CN's zone sshd starts up fine and can log in without GSS
 * Verified can log in with GSS
 * Verified can SSH out from test CN's zone to another machine with GSS login
 * Verified that key auth as root using the PubKeyPlugin for smartlogin still works
 * Did `ssh-keygen -f foobar`, verified that with no passphrase we get a PEM key, with a passphrase, we get OpenSSH format (this patch took some merging)
 * Checked that login audit events are still being produced by sshd and data looks sensible
 * Checked that LANG and LC_* variables are being accepted by default as expected

Todo (probably):
 * Check that SSH to a SunSSH server still works (the compat patch had some merge turds)
 * Double-check DH moduli fallback behaviour
 * Test tcp wrappers?
 * Test the per-session xauthfile patch again?

### CODE REVIEW

**Comment by bahamat** [comment]
This looks good so far.

I think that maybe we don't need to worry about tcp wrappers because of SMF? If you want it in zones you can install ssh from pkgsrc. For GZ, I'm ok with saying that SMF is preferred and tcp wrappers aren't supported. (If they work fine, but I don't feel it's necessary to expend extra effort to ensure it works when the GZ is intended to not use it).

I would still like to ensure that we can ssh to SunSSH systems. We migrated away from SunSSH quite a while ago, but I can think of at least one user that I'm pretty sure is still using SunSSH PIs. I'd like to avoid breaking them if possible.

**Comment by arekinath** [comment]
@bahamat the TCP wrappers patch was added back in originally because people piped up on the mailinglist saying they used it. It's tough to actually remove things when a PI upgrade changes a zone's sshd out from under it without any warning or ability to migrate config.

Re: SunSSH testing, do you want to check that then? I'm having trouble finding the motivation to do it personally ;) Happy to try fix it if it's broken, but I'd give it like an 85% chance it still works haha

File: openssh/Patches/0022-Add-SMF-manifest-and-method-and-install-them.patch [hunk 1]
```diff
@@ -266,16 +265,6 @@ index 00000000..e91ed553
 +	fi
 +}
 +
```
Comment (arekinath) [inline openssh/Patches/0022-Add-SMF-manifest-and-method-and-install-them.patch:267]: This probably needs to go back to how it was

Comment (arekinath) [reply openssh/Patches/0022-Add-SMF-manifest-and-method-and-install-them.patch:267]: Looks like it was  changed when the smartos-live overlay was merged in in 45e1cee06836861ef79ebe2514a19c956ee96b92, but the illumos-openssh patch set wasn't updated

**Review by bahamat — APPROVED** [review]
I've tested this using a PI built with `OpenSSH_8.6p1`, and I can successfully ssh to a PI running `Sun_SSH_1.5` using the default configuration for both.

**Pull request merged**